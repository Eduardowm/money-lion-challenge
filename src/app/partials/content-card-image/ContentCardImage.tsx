'use client'

import * as AspectRatio from '@radix-ui/react-aspect-ratio'
import Image from 'next/image'

interface ContentCardImageProps {
  source: string
  alt: string
}

function ContentCardImage({ source, alt }: ContentCardImageProps) {
  return (
    <AspectRatio.Root ratio={16 / 12}>
      <Image className="rounded" src={source} alt={alt} fill />
    </AspectRatio.Root>
  )
}
export default ContentCardImage
