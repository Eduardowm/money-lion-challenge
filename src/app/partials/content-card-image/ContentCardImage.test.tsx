import { render } from '@testing-library/react'

import ContentCardImage from './ContentCardImage'

describe('ContentCardImage', () => {
  it('renders as expected', () => {
    const { container } = render(<ContentCardImage source="/mock-source" alt="Mock alt" />)

    expect(container).toMatchSnapshot()
  })
})
