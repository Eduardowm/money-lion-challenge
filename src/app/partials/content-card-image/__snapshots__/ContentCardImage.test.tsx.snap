// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentCardImage renders as expected 1`] = `
<div>
  <div
    data-radix-aspect-ratio-wrapper=""
    style="position: relative; width: 100%; padding-bottom: 75%;"
  >
    <div
      style="position: absolute; top: 0px; right: 0px; bottom: 0px; left: 0px;"
    >
      <img
        alt="Mock alt"
        class="rounded"
        src="/mock-source"
      />
    </div>
  </div>
</div>
`;
